### ISSUE TITLE
Impossible to install via cargo

### ISSUE THREAD

**Issue by aborruso** [nushell/nushell#2866]
**Describe the bug**

When I run `cargo install nu --verbose`, I have compiling error.

How to solve it?

**To Reproduce**
1. simply running `cargo install nu --verbose`

**Expected behavior**

**Screenshots**

![image](https://user-images.githubusercontent.com/30607/103709271-79225c00-4fb2-11eb-88f0-abe11057717a.png)

```
   Compiling rust-embed-impl v5.7.0
     Running `rustc --crate-name rust_embed_impl --edition=2018 /home/aborruso/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-embed-impl-5.7.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type proc-macro --emit=dep-info,link -C prefer-dynamic -C opt-level=z -Cembed-bitcode=no -C codegen-units=1 -C metadata=e85d3f6f227765de -C extra-filename=-e85d3f6f227765de --out-dir /tmp/cargo-installz0XnRM/release/deps -L dependency=/tmp/cargo-installz0XnRM/release/deps --extern quote=/tmp/cargo-installz0XnRM/release/deps/libquote-ef8440215e0282f2.rlib --extern rust_embed_utils=/tmp/cargo-installz0XnRM/release/deps/librust_embed_utils-9e501703b7a8c3b0.rlib --extern syn=/tmp/cargo-installz0XnRM/release/deps/libsyn-74e5419c3d256976.rlib --extern walkdir=/tmp/cargo-installz0XnRM/release/deps/libwalkdir-21dd448c8904f42a.rlib --extern proc_macro --cap-lints allow`
error[E0432]: unresolved import `syn::export`
 --> /home/aborruso/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-embed-impl-5.7.0/src/lib.rs:8:11
  |
8 | use syn::{export::TokenStream2, Data, DeriveInput, Fields, Lit, Meta};
  |           ^^^^^^ could not find `export` in `syn`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `rust-embed-impl`.

Caused by:
  process didn't exit successfully: `rustc --crate-name rust_embed_impl --edition=2018 /home/aborruso/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-embed-impl-5.7.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type proc-macro --emit=dep-info,link -C prefer-dynamic -C opt-level=z -Cembed-bitcode=no -C codegen-units=1 -C metadata=e85d3f6f227765de -C extra-filename=-e85d3f6f227765de --out-dir /tmp/cargo-installz0XnRM/release/deps -L dependency=/tmp/cargo-installz0XnRM/release/deps --extern quote=/tmp/cargo-installz0XnRM/release/deps/libquote-ef8440215e0282f2.rlib --extern rust_embed_utils=/tmp/cargo-installz0XnRM/release/deps/librust_embed_utils-9e501703b7a8c3b0.rlib --extern syn=/tmp/cargo-installz0XnRM/release/deps/libsyn-74e5419c3d256976.rlib --extern walkdir=/tmp/cargo-installz0XnRM/release/deps/libwalkdir-21dd448c8904f42a.rlib --extern proc_macro --cap-lints allow` (exit code: 1)
warning: build failed, waiting for other jobs to finish...
error: failed to compile `nu v0.25.0`, intermediate artifacts can be found at `/tmp/cargo-installz0XnRM`
```

**Comment by jonathandturner** [comment]
@aborruso - what version of rustc are you using? You'll need a stable that's pretty recent, say something like 1.48 or even 1.49.

**Comment by TrevorAC99** [comment]
> @aborruso - what version of rustc are you using? You'll need a stable that's pretty recent, say something like 1.48 or even 1.49.

I received this same error in Windows using 1.48 of rustc. I'm trying again with 1.49 right now.

**Comment by aborruso** [comment]
I have rustc 1.46.0. I try to update it

**Comment by pantharshit00** [comment]
I can also reproduce this when I tried updating to v0.25

![image](https://user-images.githubusercontent.com/22195362/103709957-ca921180-4fd9-11eb-8fb4-e5c380f23aa0.png)

**Comment by aborruso** [comment]
I have used `rustup update stable` and now I have 1.49.

And I have again

```
   Compiling async-recursion v0.3.1
error[E0432]: unresolved import `syn::export`
 --> /home/aborruso/.cargo/registry/src/github.com-1ecc6299db9ec823/rust-embed-impl-5.7.0/src/lib.rs:8:11
  |
8 | use syn::{export::TokenStream2, Data, DeriveInput, Fields, Lit, Meta};
  |           ^^^^^^ could not find `export` in `syn`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
error: could not compile `rust-embed-impl`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `nu v0.25.0`, intermediate artifacts can be found at `/tmp/cargo-installIY3TFP`

Caused by:
  build failed
```

**Comment by TrevorAC99** [comment]
This may or may not be relevant but when I cloned the repository and built it using `cargo build`, it worked fine. For some reason, installing from crates.io has problems.

**Comment by jonathandturner** [comment]
We're working around this issue with https://github.com/nushell/nushell/pull/2868

Looks like the root cause is a semver-incompatible change that went out in the latest `syn` crate. We're currently going to pin to an older version and release. That should at least unblock the builds for everyone.

**Comment by jonathandturner** [comment]
This should be fixed with 0.25.1. If you're still seeing it, please reopen.
